--- sqlite3-3.9.1.orig/Makefile.in
+++ sqlite3-3.9.1/Makefile.in
@@ -597,6 +597,7 @@ sqlite3.pc: $(TOP)/sqlite3.pc.in
 libsqlite3.la:	$(LIBOBJ)
 	$(LTLINK) -no-undefined -o $@ $(LIBOBJ) $(TLIBS) \
 		${ALLOWRELEASE} -rpath "$(libdir)" -version-info "8:6:8"
+	sed -i "/dependency_libs/s/'.*'/''/" $@
 
 libtclsqlite3.la:	tclsqlite.lo libsqlite3.la
 	$(LTLINK) -no-undefined -o $@ tclsqlite.lo \
@@ -604,6 +605,7 @@ libtclsqlite3.la:	tclsqlite.lo libsqlite
 		-rpath "$(TCLLIBDIR)" \
 		-version-info "8:6:8" \
 		-avoid-version
+	sed -i "/dependency_libs/s/'.*'/''/" $@
 
 sqlite3$(TEXE):	shell.c sqlite3.c
 	$(LTLINK) $(READLINE_FLAGS) $(SHELL_OPT) -o $@ \
