Index: sqlite3/src/select.c
==================================================================
--- sqlite3/src/select.c
+++ sqlite3/src/select.c
@@ -4164,13 +4164,12 @@
 ){
   int i;
   assert( pColumn->op==TK_COLUMN );
   assert( sqlite3ExprIsConstant(pValue) );
 
-  if( !ExprHasProperty(pValue, EP_FixedCol) && sqlite3ExprAffinity(pValue)!=0 ){
-    return;
-  }
+  if( ExprHasProperty(pColumn, EP_FixedCol) ) return;
+  if( sqlite3ExprAffinity(pValue)!=0 ) return;
   if( !sqlite3IsBinary(sqlite3ExprCompareCollSeq(pConst->pParse,pExpr)) ){
     return;
   }
 
   /* 2018-10-25 ticket [cf5ed20f]
@@ -4189,13 +4188,10 @@
   pConst->apExpr = sqlite3DbReallocOrFree(pConst->pParse->db, pConst->apExpr,
                          pConst->nConst*2*sizeof(Expr*));
   if( pConst->apExpr==0 ){
     pConst->nConst = 0;
   }else{
-    if( ExprHasProperty(pValue, EP_FixedCol) ){
-      pValue = pValue->pLeft;
-    }
     pConst->apExpr[pConst->nConst*2-2] = pColumn;
     pConst->apExpr[pConst->nConst*2-1] = pValue;
   }
 }
 

Index: sqlite3/test/whereL.test
==================================================================
--- sqlite3/test/whereL.test
+++ sqlite3/test/whereL.test
@@ -142,7 +142,19 @@
 } {}
 do_execsql_test 530 {
   SELECT 200, * FROM t0, v0 WHERE t0.c0 = 0 AND v0.c0 = t0.c0;
 } {}
 
+# 2020-02-13: ticket 1dcb4d44964846ad
+# A problem introduced while making optimizations on the fixes above.
+#
+reset_db
+do_execsql_test 600 {
+  CREATE TABLE t1(x TEXT);
+  CREATE TABLE t2(y TEXT);
+  INSERT INTO t1 VALUES('good'),('bad');
+  INSERT INTO t2 VALUES('good'),('bad');
+  SELECT * FROM t1 JOIN t2 ON x=y
+   WHERE x='good' AND y='good';
+} {good good}
 
 finish_test

