From: Lennart Weller <lhw@ring0.de>
Date: Sat, 28 Jan 2017 17:57:59 +0100
Subject: linked js/css/fonts removed from make

---
 python.d/Makefile.am | 42 ------------------------------------------
 web/Makefile.am      | 35 -----------------------------------
 2 files changed, 77 deletions(-)

diff --git a/python.d/Makefile.am b/python.d/Makefile.am
index 883f06c..a163fe7 100644
--- a/python.d/Makefile.am
+++ b/python.d/Makefile.am
@@ -53,45 +53,3 @@ dist_pythonmodules_DATA = \
     python_modules/msg.py \
     python_modules/lm_sensors.py \
     $(NULL)
-
-pythonyaml2dir=$(pythonmodulesdir)/pyyaml2
-dist_pythonyaml2_DATA = \
-    python_modules/pyyaml2/__init__.py \
-    python_modules/pyyaml2/composer.py \
-    python_modules/pyyaml2/constructor.py \
-    python_modules/pyyaml2/cyaml.py \
-    python_modules/pyyaml2/dumper.py \
-    python_modules/pyyaml2/emitter.py \
-    python_modules/pyyaml2/error.py \
-    python_modules/pyyaml2/events.py \
-    python_modules/pyyaml2/loader.py \
-    python_modules/pyyaml2/nodes.py \
-    python_modules/pyyaml2/parser.py \
-    python_modules/pyyaml2/reader.py \
-    python_modules/pyyaml2/representer.py \
-    python_modules/pyyaml2/resolver.py \
-    python_modules/pyyaml2/scanner.py \
-    python_modules/pyyaml2/serializer.py \
-    python_modules/pyyaml2/tokens.py \
-    $(NULL)
-
-pythonyaml3dir=$(pythonmodulesdir)/pyyaml3
-dist_pythonyaml3_DATA = \
-    python_modules/pyyaml3/__init__.py \
-    python_modules/pyyaml3/composer.py \
-    python_modules/pyyaml3/constructor.py \
-    python_modules/pyyaml3/cyaml.py \
-    python_modules/pyyaml3/dumper.py \
-    python_modules/pyyaml3/emitter.py \
-    python_modules/pyyaml3/error.py \
-    python_modules/pyyaml3/events.py \
-    python_modules/pyyaml3/loader.py \
-    python_modules/pyyaml3/nodes.py \
-    python_modules/pyyaml3/parser.py \
-    python_modules/pyyaml3/reader.py \
-    python_modules/pyyaml3/representer.py \
-    python_modules/pyyaml3/resolver.py \
-    python_modules/pyyaml3/scanner.py \
-    python_modules/pyyaml3/serializer.py \
-    python_modules/pyyaml3/tokens.py \
-    $(NULL)
diff --git a/web/Makefile.am b/web/Makefile.am
index 396cf0b..e14cc57 100644
--- a/web/Makefile.am
+++ b/web/Makefile.am
@@ -4,10 +4,6 @@
 MAINTAINERCLEANFILES= $(srcdir)/Makefile.in
 
 dist_web_DATA = \
-	demo.html \
-	demo2.html \
-	demosites.html \
-	demosites2.html \
 	dashboard.html \
 	dashboard.js \
 	dashboard_info.js \
@@ -25,15 +21,6 @@ dist_web_DATA = \
 	version.txt \
 	$(NULL)
 
-webolddir=$(webdir)/old
-dist_webold_DATA = \
-	old/datasource.html \
-	old/index.html \
-	old/index.js \
-	old/netdata.js \
-	old/theme.css \
-	$(NULL)
-
 weblibdir=$(webdir)/lib
 dist_weblib_DATA = \
 	lib/bootstrap-3.3.7.min.js \
@@ -62,23 +49,11 @@ dist_webcss_DATA = \
 	css/bootstrap-theme-3.3.7.min.css \
 	css/bootstrap-slate-flat-3.3.7.css \
 	css/bootstrap-toggle-2.2.2.min.css \
-	css/font-awesome.min.css \
 	css/c3-0.4.11.min.css \
 	$(NULL)
 
 webfontsdir=$(webdir)/fonts
 dist_webfonts_DATA = \
-	fonts/glyphicons-halflings-regular.eot \
-	fonts/glyphicons-halflings-regular.svg \
-	fonts/glyphicons-halflings-regular.ttf \
-	fonts/glyphicons-halflings-regular.woff \
-	fonts/glyphicons-halflings-regular.woff2 \
-	fonts/FontAwesome.otf \
-	fonts/fontawesome-webfont.eot \
-	fonts/fontawesome-webfont.svg \
-	fonts/fontawesome-webfont.ttf \
-	fonts/fontawesome-webfont.woff \
-	fonts/fontawesome-webfont.woff2 \
 	$(NULL)
 
 webimagesdir=$(webdir)/images
@@ -105,16 +80,6 @@ dist_webimages_DATA = \
 	images/seo-performance-multi-size.icns \
 	$(NULL)
 
-
-webwellknowndir=$(webdir)/.well-known
-dist_webwellknown_DATA = \
-	$(NULL)
-
-webdntdir=$(webdir)/.well-known/dnt
-dist_webdnt_DATA = \
-	.well-known/dnt/cookies \
-	$(NULL)
-
 version.txt:
 	if test -d "$(top_srcdir)/.git"; then \
 		git --git-dir="$(top_srcdir)/.git" log -n 1 --format=%H; \
