From: Sven Hartge <sven@svenhartge.de>
Date: Sun, 26 Feb 2017 23:57:41 +0100
Subject: Debianize path to exim4 binary

---
 conf.d/python.d/exim.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/conf.d/python.d/exim.conf b/conf.d/python.d/exim.conf
index 07d72c5..dd2663d 100644
--- a/conf.d/python.d/exim.conf
+++ b/conf.d/python.d/exim.conf
@@ -83,4 +83,4 @@ update_every: 10
 # AUTO-DETECTION JOBS
 
 local:
-  command: 'exim -bpc'
+  command: '/usr/sbin/exim4 -bpc'
-- 
2.11.0
