diff -ruN python-eventlet-0.20.0/eventlet/wsgi.py python-eventlet-0.20.0-patched/eventlet/wsgi.py
--- python-eventlet-0.20.0/eventlet/wsgi.py	2016-12-11 22:09:40.000000000 +0100
+++ python-eventlet-0.20.0-patched/eventlet/wsgi.py	2019-10-07 18:07:40.001424543 +0200
@@ -627,7 +627,7 @@
             else:
                 env[envk] = v
 
-        if env.get('HTTP_EXPECT') == '100-continue':
+        if env.get('HTTP_EXPECT', '').lower() == '100-continue':
             wfile = self.wfile
             wfile_line = b'HTTP/1.1 100 Continue\r\n'
         else:
