--- diamond-20160916.git175a4db.orig/setup.py
+++ diamond-20160916.git175a4db/setup.py
@@ -60,20 +60,11 @@ else:
         data_files.append(('/etc/diamond/handlers',
                            glob('conf/handlers/*')))
 
-        if distro == 'Ubuntu':
-            data_files.append(('/etc/init',
-                               ['debian/diamond.upstart']))
-        if distro in ['centos', 'redhat', 'debian', 'fedora', 'oracle']:
-            data_files.append(('/etc/init.d',
-                               ['bin/init.d/diamond']))
+        if distro in ['Ubuntu','centos', 'redhat', 'debian', 'fedora', 'oracle']:
             data_files.append(('/var/log/diamond',
                                ['.keep']))
-            if distro_major_version >= 7 and not distro == 'debian':
-                data_files.append(('/usr/lib/systemd/system',
-                                   ['rpm/systemd/diamond.service']))
-            elif distro_major_version >= 6 and not distro == 'debian':
-                data_files.append(('/etc/init',
-                                   ['rpm/upstart/diamond.conf']))
+            data_files.append(('/usr/lib/systemd/system',
+                               ['rpm/systemd/diamond.service']))
 
     # Support packages being called differently on different distros
 
