diff --git a/src/c/configure.ac b/src/c/configure.ac
index be00af9..8193511 100644
--- a/src/c/configure.ac
+++ b/src/c/configure.ac
@@ -34,7 +34,7 @@ if test "$with_cppunit" = "no" ; then
    CPPUNIT_INCLUDE=
    CPPUNIT_LIBS=
 else
-   AM_PATH_CPPUNIT(1.10.2)
+   PKG_CHECK_MODULES(CPPUNIT, cppunit >= 1.10.2)
 fi
 
 if test "$CALLER" = "ANT" ; then
diff --git a/src/recipes/lock/src/c/configure.ac b/src/recipes/lock/src/c/configure.ac
index 3954123..bc86461 100644
--- a/src/recipes/lock/src/c/configure.ac
+++ b/src/recipes/lock/src/c/configure.ac
@@ -72,7 +72,7 @@ AC_C_VOLATILE
 AC_PROG_CC
 AC_PROG_LIBTOOL
 #check for cppunit 
-AM_PATH_CPPUNIT(1.10.2)
+PKG_CHECK_MODULES(CPPUNIT, cppunit >= 1.10.2)
 # Checks for library functions.
 AC_FUNC_UTIME_NULL
 AC_CHECK_FUNCS([gettimeofday memset mkdir rmdir strdup strerror strstr strtol strtoul strtoull utime])
diff --git a/src/recipes/queue/src/c/configure.ac b/src/recipes/queue/src/c/configure.ac
index a9fb7b1..32fdf94 100644
--- a/src/recipes/queue/src/c/configure.ac
+++ b/src/recipes/queue/src/c/configure.ac
@@ -72,7 +72,7 @@ AC_C_VOLATILE
 AC_PROG_CC
 AC_PROG_LIBTOOL
 #check for cppunit 
-AM_PATH_CPPUNIT(1.10.2)
+PKG_CHECK_MODULES(CPPUNIT, cppunit >= 1.10.2)
 # Checks for library functions.
 AC_FUNC_UTIME_NULL
 AC_CHECK_FUNCS([gettimeofday memset mkdir rmdir strdup strerror strstr strtol strtoul strtoull utime])
