Index: zookeeper/src/java/test/org/apache/zookeeper/server/quorum/auth/MiniKdcTest.java
===================================================================
--- zookeeper.orig/src/java/test/org/apache/zookeeper/server/quorum/auth/MiniKdcTest.java
+++ zookeeper/src/java/test/org/apache/zookeeper/server/quorum/auth/MiniKdcTest.java
@@ -21,6 +21,7 @@ package org.apache.zookeeper.server.quor
 import org.apache.directory.server.kerberos.shared.keytab.Keytab;
 import org.apache.directory.server.kerberos.shared.keytab.KeytabEntry;
 import org.junit.Assert;
+import org.junit.Ignore;
 import org.junit.Test;
 
 import javax.security.auth.Subject;
@@ -47,12 +48,14 @@ public class MiniKdcTest extends Kerbero
     private static final boolean IBM_JAVA = System.getProperty("java.vendor")
             .contains("IBM");
 
+    @Ignore("disabled on Debian builds")
     @Test(timeout = 60000)
     public void testMiniKdcStart() {
         MiniKdc kdc = getKdc();
         Assert.assertNotSame(0, kdc.getPort());
     }
 
+    @Ignore("disabled on Debian builds")
     @Test(timeout = 60000)
     public void testKeytabGen() throws Exception {
         MiniKdc kdc = getKdc();
@@ -131,6 +134,7 @@ public class MiniKdcTest extends Kerbero
         }
     }
 
+    @Ignore("disabled on Debian builds")
     @Test(timeout = 60000)
     public void testKerberosLogin() throws Exception {
         MiniKdc kdc = getKdc();
Index: zookeeper/src/java/test/org/apache/zookeeper/server/quorum/auth/QuorumKerberosHostBasedAuthTest.java
===================================================================
--- zookeeper.orig/src/java/test/org/apache/zookeeper/server/quorum/auth/QuorumKerberosHostBasedAuthTest.java
+++ zookeeper/src/java/test/org/apache/zookeeper/server/quorum/auth/QuorumKerberosHostBasedAuthTest.java
@@ -35,6 +35,7 @@ import org.apache.zookeeper.test.ClientB
 import org.junit.After;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
+import org.junit.Ignore;
 import org.junit.Test;
 
 import junit.framework.Assert;
@@ -122,6 +123,7 @@ public class QuorumKerberosHostBasedAuth
     /**
      * Test to verify that server is able to start with valid credentials
      */
+    @Ignore("disabled on Debian builds")
     @Test(timeout = 120000)
     public void testValidCredentials() throws Exception {
         String serverPrincipal = hostServerPrincipal.substring(0, hostServerPrincipal.lastIndexOf("@"));
@@ -143,6 +145,7 @@ public class QuorumKerberosHostBasedAuth
     /**
      * Test to verify that the bad server connection to the quorum should be rejected.
      */
+    @Ignore("disabled on Debian builds")
     @Test(timeout = 120000)
     public void testConnectBadServer() throws Exception {
         String serverPrincipal = hostServerPrincipal.substring(0, hostServerPrincipal.lastIndexOf("@"));
Index: zookeeper/src/java/test/org/apache/zookeeper/server/quorum/auth/QuorumKerberosAuthTest.java
===================================================================
--- zookeeper.orig/src/java/test/org/apache/zookeeper/server/quorum/auth/QuorumKerberosAuthTest.java
+++ zookeeper/src/java/test/org/apache/zookeeper/server/quorum/auth/QuorumKerberosAuthTest.java
@@ -33,6 +33,7 @@ import org.apache.zookeeper.test.ClientB
 import org.junit.After;
 import org.junit.AfterClass;
 import org.junit.Before;
+import org.junit.Ignore;
 import org.junit.Test;
 
 public class QuorumKerberosAuthTest extends KerberosSecurityTestcase {
@@ -94,6 +95,7 @@ public class QuorumKerberosAuthTest exte
     /**
      * Test to verify that server is able to start with valid credentials
      */
+    @Ignore("disabled on Debian builds")
     @Test(timeout = 120000)
     public void testValidCredentials() throws Exception {
         String serverPrincipal = KerberosTestUtils.getServerPrincipal();
